stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -238,12 +402,20 @@ class cholesky_scalar : public vari {
 inline Eigen::Matrix<var, -1, -1> cholesky_decompose(
     const Eigen::Matrix<var, -1, -1>& A) {
   check_square("cholesky_decompose", "A", A);
-  check_symmetric("cholesky_decompose", "A", A);
```

Comment:
I moved the check_symmetric to the CPU versions as in the OpenCL version, this is actually duplicated (this check happens in `prim`). I left check_square as is, as it does not affect performance.

Reply:
Great catch!